### PR TITLE
Add SN5640 SIMX platform with default SKU

### PR DESCRIPTION
**Pull request by noaOrMlnx**
#### What I did
Added new platform and SKU for SN5640. 
Currently, number of up/down links is 256/256. 

##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2 [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+#}
+{% set default_cable = '300m' %}
+{% set ingress_lossless_pool_size =  '31502336' %}
+{% set ingress_lossy_pool_size =  '31502336' %}
+{% set egress_lossless_pool_size =  '138240000' %}
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2:20]: I am afraid this value is too bug. if the guaranteed size is 130.9, the number should be round_down(130.9*1024*1024, 192)

File: device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers.json.j2 [hunk 2]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/buffers.json.j2
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers.json.j2:1]: ../../x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers.json.j2
```suggestion
../../x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers.json.j2
```

File: device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2 [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+#}
+{% set default_cable = '40m' %}
+{% set ingress_lossless_pool_size =  '2359296' %}
+{% set ingress_lossy_pool_size =  '2359296' %}
+{% set egress_lossless_pool_size =  '136209408' %}
+{% set egress_lossy_pool_size =  '2359296' %}
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2:21]:
```suggestion
{% set egress_lossy_pool_size =  '2359296' %}
{% set ingress_lossless_pool_xoff =  '1024000' %}
```

File: device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t1.j2 [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+#}
+{% set default_cable = '300m' %}
+{% set ingress_lossless_pool_size =  '18874368' %}
+{% set ingress_lossy_pool_size =  '71459840' %}
+{% set egress_lossless_pool_size =  '136209408' %}
+{% set egress_lossy_pool_size =  '71459840' %}
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t1.j2:21]:
```suggestion
{% set egress_lossy_pool_size =  '71459840' %}
{% set ingress_lossless_pool_xoff =  '1024000' %}
```

**Review by stephenxs — COMMENTED** [review]
Just to overcome your issue.
Need to understand the number of uplink/downlink ports to calculate concrete pool numbers

File: device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2 [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+#}
+{% set default_cable = '40m' %}
+{% set ingress_lossless_pool_size =  '73684992' %}
+{% set ingress_lossless_pool_xoff =  '57828352' %}
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5640_simx-r0/ACS-SN5640/buffers_defaults_t0.j2:19]: 22016 * 1024?

**Comment by noaOrMlnx** [comment]
@stephenxs thanks for the comment. all were addressed.

**Review by stephenxs — APPROVED** [review]

**Pull request closed without merging**